netcontrold/lib/util.py

```diff
@@ -63,10 +63,17 @@ def cpuinfo():
         regex = re.compile('^(.*?)\s*:\s*(.*)')
         (param, val) = regex.match(line).groups()
 
+        if val=='':
+            continue
+
+        if param == 'processor' and val=='':
```

Comment:
However, empty val should not be ignored when the field is "processor" as it is the key for processor lookup later.

You raise an exception when val is '' (i.e nothing). Then, catch this exception in to assert as expected. We should not ignore this case.  Suggestion: create new class 'UtilExc' in error.py and raise here. calling function(s) will handle it.
Good check!..